### PR TITLE
Added danger gem command.

### PR DESCRIPTION
**Pull request by dblock**
Added a `danger gem` command that creates a new `dangerfile-` gem using https://github.com/danger/dangerfile-gem-template. This closes 
https://github.com/danger/danger.systems/issues/106.

### CODE REVIEW

**Comment by orta** [comment]
Hrm, I'm not sure this is a good idea - this is basically a C&P of https://github.com/dbgrandi/claide-plugins/blob/claide-plugins/lib/claide/command/plugins/create.rb

We should do it properly by making the create command from claide work on both

**Comment by dblock** [comment]
Claid is a CocoaPods plugin which shows info about available CocoaPods plugins or helps you get started developing a new plugin. It has been reused in Danger but not updated to be a generic thing. I don't think I am motivated enough to go down the rabbit hole of refactoring that thing to add a command to danger :)

**Comment by dblock** [comment]
This test that runs the plugin fails on Windows. AFAIK `danger plugin` will fail on Windows too. What should I do about that?

**Comment by orta** [comment]
Only the README hasn't been changed, inside it's 100% generic - we've just not taken those changes back to CocoaPods, might never end up doing that - hard to say. 

WRT windows, not really sure about that, if it doesn't have fork we'd have to move to `system` maybe?

**Comment by orta** [comment]
Hrm, it is using `system` - that sucks

**Comment by dblock** [comment]
I've conditioned the test not to run on Windows and opened https://github.com/dbgrandi/claide-plugins/issues/4.

**Comment by dblock** [comment]
If someone refactors claide-plugins to be reusable, I'll gladly update this PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+* Add `danger gem` - dblock
 * Link to https://danger.systems/ in Bitbucket Server comments - HeEAaD
 * Add `markdown_link` for the Bitbucket Server plugin - HeEAaD
 * Updated Bitrise documentation about env vars - kylejm
 
 ## 3.2.1
 
-* Fix for markdown accidentally crashing Danger when sending a message #520 - orta 
-* Improved the width handling for the output of the `danger local` table - orta 
-* Added a check for the merge inflection point inside the diff, and will fetch it if needed - orta
```
Comment (k0nserv) [inline CHANGELOG.md]: These spaces should stick around right?

Comment (dblock) [inline CHANGELOG.md]: My editor removed them. I can do surgery to put them back, but why? :)

Comment (JuanitoFatas) [inline CHANGELOG.md]: Why [we exclude markdown files not trim trailing whitespaces](https://github.com/danger/danger/blob/master/.editorconfig#L16-L17)? :/

Comment (dblock) [inline CHANGELOG.md]: I love me a good tabs vs. spaces debate :) +1 @JuanitoFatas to make a PR to remove that.

Comment (orta) [inline CHANGELOG.md]: It's CocoaPods' fault - the CHANGELOG there is whitespace EOL dependent and so I try keep some constency

Comment (orta) [inline CHANGELOG.md]: Personally I don't care, it's meant to be human readable, happy to have them removed etc

**Comment by orta** [comment]
I've made the changes in claide-plugins for this

**Comment by orta** [comment]
I've shipped claide-plugins 0.9.2 with the API for this now - https://github.com/dbgrandi/claide-plugins

Will try and re-create this PR with that now in the next week

**Comment by orta** [comment]
@dbgrandi ^

**Comment by orta** [comment]
Awesome, I've taken this and made #608 with a bit more infrastructure 👍

**Pull request closed without merging**